stellar-multitenant/pkg/cli/add_tenants.go

```diff
@@ -169,27 +179,52 @@ func executeAddTenant(
 	tenantName, userFirstName, userLastName, userEmail, organizationName string,
 	messengerClient message.MessengerClient,
 	tenantsOpts AddTenantsCommandOptions,
-	sigOpts signing.SignatureServiceOptions,
+	txSubOpts di.TxSubmitterEngineOptions,
+	distAccResolverOpts signing.DistributionAccountResolverOptions,
 ) error {
-	if !slices.Contains(signing.DistributionSignatureClientTypes(), sigOpts.DistributionSignerType) {
-		return fmt.Errorf("invalid distribution account signer type %q", sigOpts.DistributionSignerType)
+	if !slices.Contains(signing.DistributionSignatureClientTypes(), txSubOpts.SignatureServiceOptions.DistributionSignerType) {
+		return fmt.Errorf("invalid distribution account signer type %q", txSubOpts.SignatureServiceOptions.DistributionSignerType)
+	}
+	txSubOpts.SignatureServiceOptions.DBConnectionPool = tssDBConnectionPool
+
+	horizonClient := &horizonclient.Client{
+		HorizonURL: txSubOpts.HorizonURL,
+		HTTP:       httpclient.DefaultClient(),
 	}
 
-	distAccSigClient, err := signing.NewSignatureClient(sigOpts.DistributionSignerType, signing.SignatureClientOptions{
-		NetworkPassphrase:           sigOpts.NetworkPassphrase,
-		DistributionPrivateKey:      sigOpts.DistributionPrivateKey,
-		DistAccEncryptionPassphrase: sigOpts.DistAccEncryptionPassphrase,
-		DBConnectionPool:            tssDBConnectionPool,
+	distAccResolver, err := signing.NewDistributionAccountResolver(signing.DistributionAccountResolverOptions{
+		AdminDBConnectionPool:            adminDBConnectionPool,
+		HostDistributionAccountPublicKey: distAccResolverOpts.HostDistributionAccountPublicKey,
 	})
 	if err != nil {
-		return fmt.Errorf("creating a new distribution account signature client: %w", err)
+		return fmt.Errorf("creating a new distribution account resolver instance: %w", err)
+	}
+	txSubOpts.SignatureServiceOptions.DistributionAccountResolver = distAccResolver
+
+	ledgerNumberTracker, err := preconditions.NewLedgerNumberTracker(horizonClient)
+	if err != nil {
+		return fmt.Errorf("grabbing ledger number tracker instance: %w", err)
+	}
+	txSubOpts.SignatureServiceOptions.LedgerNumberTracker = ledgerNumberTracker
+
+	signatureService, err := signing.NewSignatureService(txSubOpts.SignatureServiceOptions)
+	if err != nil {
+		return fmt.Errorf("grabbing signature service instance: %w", err)
+	}
+
+	txSubmitterEngine := engine.SubmitterEngine{
```

Comment:
Why are we manually building this ? Can't we just use `submitterEngine, err := di.NewTxSubmitterEngine(ctx, txSubmitterOpts)` that gets you a singleton of the submitter engine ?

Reply:
we wouldn't want to use the dependency injected singleton since the cli allows users to customize the values for the configs that are used to build the services which may differ from the configs that were defined at startup when serving the api.